README.md

```diff
@@ -1,707 +1,1825 @@
+## Additional Tips & Tricks
 
-```js
-function defaultEqualityCheck(previousVal, currentVal) {
-  return currentVal === previousVal
+<details><summary><b>Top Level Selectors Pattern</b></summary>
+
+This pattern simplifies the creation of selectors in your application. Start by defining a generic type for your top-level selectors:
+
+```ts
+import type { Selector } from 'reselect'
+
+export type TopLevelSelectors<State> = {
+  [K in keyof State as K extends string
+    ? `select${Capitalize<K>}`
+    : never]: Selector<State, State[K], never>
 }
 ```
 
-As of Reselect 4.1, `defaultMemoize` also accepts an options object as its first argument instead of `equalityCheck`. The options object may contain:
+With this setup, you can easily create top-level selectors for your application's state:
 
 ```ts
-interface DefaultMemoizeOptions {
-  equalityCheck?: EqualityFn
-  resultEqualityCheck?: EqualityFn
-  maxSize?: number
+const topLevelSelectors: TopLevelSelectors<RootState> = {
+  selectAlerts: state => state.alerts,
+  selectTodos: state => state.todos,
+  selectUsers: state => state.users
 }
 ```
 
-Available options are:
+This approach allows for more streamlined and readable selector creation. For example:
 
-- `equalityCheck`: used to compare the individual arguments of the provided calculation function
-- `resultEqualityCheck`: if provided, used to compare a newly generated output value against previous values in the cache. If a match is found, the old value is returned. This address the common `todos.map(todo => todo.id)` use case, where an update to another field in the original data causes a recalculate due to changed references, but the output is still effectively the same.
-- `maxSize`: the cache size for the selector. If `maxSize` is greater than 1, the selector will use an LRU cache internally
+```ts
+const selectCompletedTodos = createSelector(
+  [topLevelSelectors.selectTodos],
+  todos => todos.filter(todo => todo.completed === true)
+)
+```
 
-The returned memoized function will have a `.clearCache()` method attached.
+</details>
 
-`defaultMemoize` can also be used with `createSelectorCreator` to create a new selector factory that always has the same settings for each selector.
+<div align="right">[ <a href="#table-of-contents">↑ Back to top ↑</a> ]</div>
 
-### createSelectorCreator(memoize, ...memoizeOptions)
+---
 
-`createSelectorCreator` can be used to make a customized version of `createSelector`.
+## FAQ
 
-The `memoize` argument is a memoization function to replace `defaultMemoize`.
+### Why isn’t my selector recomputing when the input state changes?
 
-The `...memoizeOptions` rest parameters are zero or more configuration options to be passed to `memoizeFunc`. The selectors `resultFunc` is passed as the first argument to `memoize` and the `memoizeOptions` are passed as the second argument onwards:
+<details><summary><b>Answer</b></summary>
```

Comment:
:hand: Actually going to remove most of these `<details>` sections from the FAQs, except for maybe a couple longer answers